code/notificationbuilder/src/test/java/com/adobe/marketing/mobile/notificationbuilder/internal/builders/ManualCarouselNotificationBuilderTest.kt

```diff
@@ -353,4 +459,28 @@ class ManualCarouselNotificationBuilderTest {
             )
         }
     }
+
+    @Test
+    fun `test setupNavigationButtons for DEFAULT_MANUAL_CAROUSEL_MODE`() {
+        val mockBundle = MockCarousalTemplateDataProvider.getMockedBundleWithManualCarouselData()
+        mockBundle.putString(PushTemplateConstants.PushPayloadKeys.CAROUSEL_LAYOUT, PushTemplateConstants.DefaultValues.DEFAULT_MANUAL_CAROUSEL_MODE)
+        val data = IntentData(mockBundle, PushTemplateConstants.IntentActions.MANUAL_CAROUSEL_RIGHT_CLICKED)
+        val mcPushTemplate = CarouselPushTemplate(data) as ManualCarouselPushTemplate
+
+        val pendingIntent = mockk<PendingIntent>()
+
+        mockkConstructor(ManualCarouselNotificationBuilder::class)
+
+        every {
+            anyConstructed<ManualCarouselNotificationBuilder>().createCarouselNavigationClickPendingIntent(
+                context, mcPushTemplate, any(), broadcastReceiverClass, "channelId"
+            )
+        } returns pendingIntent
```

Comment:
Why are we mocking `createCarouselNavigationClickPendingIntent`? Can we instead mock `AEPPushNotificationBuilder.createIntent` to return a mocked intent and verify that the correct center image index gets added to it?
Additional we can also verify if the PendingIntent returned by this method resolved to the broadcast receiver class

Reply:
1. Now we're mocking `AEPPushNotificationBuilder.createIntent` rather than mocking `createCarouselNavigationClickPendingIntent`
2. Since we're not building the ManualCarouselNotificationBuilder it wouldn't make sense to test the center index in this case
3. Because now we're mocking the `AEPPushNotificationBuilder.createIntent` we can't verify the broadcast receiver class here for this test